### PR TITLE
JNI support for Collect Ops in Reduction

### PR DESCRIPTION
**Pull request by sperlingxx**
Exposes public APIs for collect operations as `ReductionAggregation`, which are essential to spark-rapids.  In addition, this PR also extends the test framework of Reduction to discriminate output types from input types.

### CODE REVIEW

File: java/src/test/java/ai/rapids/cudf/ReductionTest.java [hunk 1]
```diff
@@ -77,186 +81,356 @@ private static Scalar buildExpectedScalar(ReductionAggregation op, DType baseTyp
...
     }
   }
 
   @ParameterizedTest
   @MethodSource("createBooleanParams")
-  void testBoolean(ReductionAggregation op, Boolean[] values, Object expectedObject, Double delta) {
-    try (Scalar expected = buildExpectedScalar(op, DType.BOOL8, expectedObject);
+  void testBoolean(ReductionAggregation op, Boolean[] values,
+      HostColumnVector.DataType tpe, Object expectedObject, Double delta) {
+    try (Scalar expected = buildExpectedScalar(op, tpe, expectedObject);
```
Comment (jlowe) [inline java/src/test/java/ai/rapids/cudf/ReductionTest.java:433]: Are all of the occurrences of `tpe` supposed to be `type`?

**Review by jlowe — APPROVED** [review]
Small nits but otherwise lgtm.

File: java/src/test/java/ai/rapids/cudf/ReductionTest.java (hunk 1 above)
Comment (sperlingxx) [reply java/src/test/java/ai/rapids/cudf/ReductionTest.java]: Done.

File: java/src/main/java/ai/rapids/cudf/ReductionAggregation.java [hunk 2]
```diff
@@ -209,4 +209,67 @@ public static ReductionAggregation nth(int offset) {
...
+  /**
+   * Merge the partial lists produced by multiple CollectListAggregations.
+   * NOTICE: The partial lists to be merged should NOT include any null list element (but can include null list entries).
+   */
+  public static ReductionAggregation mergeLists() {
+    return new ReductionAggregation(Aggregation.mergeLists());
+  }
+
+  /**
+   * Merge the partial sets produced by multiple CollectSetAggregations. Each null/nan value will be regarded as
```
[the remark this answers is not in the document]
Comment (sperlingxx) [reply java/src/main/java/ai/rapids/cudf/ReductionAggregation.java]: Fixed.

File: java/src/main/java/ai/rapids/cudf/ReductionAggregation.java [hunk 3]
```diff
@@ -209,4 +209,67 @@ public static ReductionAggregation nth(int offset) {
...
+   * Collect the values into a list.
+   *
+   * @param nullPolicy Indicates whether to include/exclude nulls during collection.
+   */
+  public static ReductionAggregation collectList(NullPolicy nullPolicy) {
+    return new ReductionAggregation(Aggregation.collectList(nullPolicy));
+  }
+
+  /**
+   * Collect the values into a set. All null values will be excluded, and all nan values are regarded as
```
[the remark this answers is not in the document]
Comment (sperlingxx) [reply java/src/main/java/ai/rapids/cudf/ReductionAggregation.java]: Fixed,

**Comment by sperlingxx** [comment]
@gpucibot merge

**Pull request merged**